test/gov/hzn_sdo.sh

```diff
@@ -0,0 +1,1166 @@
+rc=$?
+if [[ $rc -eq 1 && "$cmdOutput" == *'Error:'*'exchange user authentication must be specified'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys remove' without HZN_EXCHANGE_USER_AUTH set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_EXCHANGE_USER_AUTH="$USERDEV_ADMIN_AUTH"
+
+echo -e "${PREFIX} Testing 'hzn sdo keys remove' with incorrect HZN_EXCHANGE_USER_AUTH set"
+export HZN_EXCHANGE_USER_AUTH=fakeuser:fakepw
+cmdOutput=$(hzn sdo keys remove test-sdo-key 2>&1)
+rc=$?
+if [[ $rc -eq 7 && "$cmdOutput" == *'Error:'*'Invalid credentials.'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys remove' with incorrect HZN_EXCHANGE_USER_AUTH set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_EXCHANGE_USER_AUTH=$USERDEV_ADMIN_AUTH
+
+echo -e "${PREFIX} Testing 'hzn sdo keys remove' without HZN_ORG_ID set"
+unset HZN_ORG_ID
+cmdOutput=$(hzn sdo keys remove test-sdo-key 2>&1)
+rc=$?
+if [[ $rc -eq 1 && "$cmdOutput" == *'Error:'*'organization ID must be specified'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys remove' with incorrect HZN_ORG_ID set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_ORG_ID=$HZN_ORG_ID_SAVE
+
+echo -e "${PREFIX} Testing 'hzn sdo keys remove' with incorrect HZN_ORG_ID set"
+export HZN_ORG_ID=fakeorg
+cmdOutput=$(hzn sdo keys remove test-sdo-key 2>&1)
+rc=$?
+if [[ $rc -eq 7 && "$cmdOutput" == *'Error:'*'Invalid credentials.'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys remove' without HZN_ORG_ID set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_ORG_ID=$HZN_ORG_ID_SAVE
+
+echo -e "${PREFIX} Testing 'hzn sdo keys remove ' without HZN_SDO_SVC_URL set"
+unset HZN_SDO_SVC_URL
+cmdOutput=$(hzn sdo keys remove test-sdo-key 2>&1)
+rc=$?
+if [[ $rc -eq 7 && "$cmdOutput" == *'Error:'*'Could not get'*'HZN_SDO_SVC_URL'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys remove' without HZN_SDO_SVC_URL set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_SDO_SVC_URL=$HZN_SDO_SVC_URL_SAVE
+
+echo -e "${PREFIX} Testing 'hzn sdo keys remove <keyName>'"
+cmdOutput=$(hzn sdo keys remove test-sdo-key 2>&1)
+rc=$?
+if [[ $rc -eq 0 && "$cmdOutput" == *'Key'*'successfully deleted from the SDO owner services.'* ]]; then
+	echo -e "${PREFIX} completed."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys remove': exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+
+# echo -e "${PREFIX} Testing 'hzn sdo keys remove <keyName>' when key does not exist"
```

Comment:
Why is this test case commented out?

Reply:
I did that because the API updates were not reflected in the test environment which would cause these tests to fail. They have now been updated and uncommented.